packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -235,13 +242,22 @@ const Injector = ({ children }: { children: React.ReactNode }): JSX.Element => {
         const ethProvider = new ethers.providers.JsonRpcProvider(
           partnerNetwork.url
         )
-        return ethProvider.getSigner(usersMetamaskAddress!)
+        return ethProvider.getSigner(voidSignerAddress || usersMetamaskAddress!)
+      }
+      if (voidSignerAddress) {
+        const ethProvider = new ethers.providers.JsonRpcProvider(network.url)
+        return ethProvider.getSigner(voidSignerAddress)
       }
+
       return library?.getSigner(0) as JsonRpcSigner
     }
 
     function getL2Signer(network: Network) {
       if (network.isArbitrum) {
+        if (voidSignerAddress) {
+          const arbProvider = new ethers.providers.JsonRpcProvider(network.url)
+          return arbProvider.getSigner(voidSignerAddress)
```

Comment:
I thought here we would need to `(new ethers.VoidSigner(voidSignerAddress)).connect(arbProvider)`